test/models/user_test.rb

```diff
@@ -32,4 +32,35 @@ class UserTest < ActiveSupport::TestCase
     )
     assert user.valid?
   end
+
+  test 'can list RBAC permission' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+
+    data = ['app:resource0:*', 'app:resource1:write']
+    current_user.stubs(:request_permission_data).returns(data)
+
+    compare_permissions = [
+      { app: { resource0: '*' } },
+      { app: { resource1: 'write' } }
+    ]
+
+    assert current_user.permission == compare_permissions
+  end
+
+  test 'can test RBAC resources authorization' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+
+    data = ['app:resource0:*', 'app:resource1:write']
+    current_user.stubs(:request_permission_data).returns(data)
+
+    assert current_user.authorized_to?('app', 'resource0', '*')
+    assert current_user.authorized_to?('app', 'resource0', 'destroy')
+    assert current_user.authorized_to?('app', 'resource1', 'write')
+    assert_not current_user.authorized_to?('app', '*', 'link')
+    assert_not current_user.authorized_to?('app', '*', '*')
+    assert_not current_user.authorized_to?('app', 'resource1', 'read')
+    assert_not current_user.authorized_to?('app', 'resource1', '*')
+  end
```

Comment:
I'd like to see a test that checks if the RBAC API is only called once across a request (caching).